Lib/test/test_posix.py

```diff
@@ -2175,30 +2175,53 @@ def test_utime(self):
 class NamespacesTests(unittest.TestCase):
     """Tests for os.unshare() and os.setns()."""
 
+    @support.requires_subprocess()
+    def subprocess(self, code):
+        import subprocess
+        with subprocess.Popen((sys.executable, '-c', code),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            encoding="utf-8"
+        ) as p:
+            p.wait()
+            return (
+                p.returncode,
+                tuple(p.stdout),
+                tuple(p.stderr),
+            )
+
     @unittest.skipUnless(hasattr(os, 'unshare'), 'needs os.unshare()')
     @unittest.skipUnless(hasattr(os, 'setns'), 'needs os.setns()')
-    @unittest.skipUnless(hasattr(os, 'readlink'), 'needs os.readlink()')
     @unittest.skipUnless(os.path.exists('/proc/self/ns/uts'), 'need /proc/self/ns/uts')
     @support.requires_linux_version(3, 0, 0)
     def test_unshare_setns(self):
-        original = os.readlink('/proc/self/ns/uts')
-        original_fd = os.open('/proc/self/ns/uts', os.O_RDONLY)
-        self.addCleanup(os.close, original_fd)
-
-        try:
-            os.unshare(os.CLONE_NEWUTS)
-        except OSError as e:
-            self.assertEqual(e.errno, errno.EPERM)
-            self.skipTest("unprivileged users cannot call unshare.")
-
-        current = os.readlink('/proc/self/ns/uts')
-        self.assertNotEqual(original, current)
+        rc, out, err = self.subprocess("""
```

Comment:
You can use this trick as a poor-mans-dedent:

```suggestion
        rc, out, err = self.subprocess("""if 1:
```

This allows you to indent the test code in the string so it's more readable.